perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3774,28 +3801,29 @@ private void updateMemberStatus(PerunSession sess, RichMember memberToUpdate) th
 	 * @param skippedMembers list of not successfully synchronized members
 	 */
 	public void addMissingMemberWhileSynchronization(PerunSession sess, Group group, CandidateSync candidate, List<String> overwriteUserAttributesList, List<String> mergeMemberAttributesList, List<String> skippedMembers) {
-		Member member;
-		try {
-			// Check if the member is already in the VO (just not in the group)
-			List<UserExtSource> list = new ArrayList<>();
-			for (RichUserExtSource richUserExtSource : candidate.getRichUserExtSources()) {
-				UserExtSource asUserExtSource = richUserExtSource.asUserExtSource();
-				list.add(asUserExtSource);
-			}
-			member = getPerunBl().getMembersManagerBl().getMemberByUserExtSources(sess, getPerunBl().getGroupsManagerBl().getVo(sess, group), list);
-
-			// member exists - update attributes
-			RichMember memberToUpdate = getPerunBl().getMembersManagerBl().getRichMember(sess, member);
-			try {
-				updateExistingMemberWhileSynchronization(sess, group, candidate, memberToUpdate, overwriteUserAttributesList, mergeMemberAttributesList, new ArrayList<>());
-			} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
-				// if update fails, skip him
-				log.warn("Can't update member from candidate {} due to attribute value exception {}.", candidate, e);
-				skippedMembers.add("MemberEntry:[" + candidate + "] was skipped because there was problem when updating member from candidate: Exception: " + e.getName() + " => '" + e.getMessage() + "'");
-				return;
+		Member member = null;
+		// Check if the member is already in the VO (just not in the group)
+		List<RichMember> richMembers = getPerunBl().getMembersManagerBl().getRichMembers(sess, getPerunBl().getGroupsManagerBl().getVo(sess, group));
```

Comment:
Doing this for every member of the group seems like really expensive operation. I am afraid we need to find better way.